packages/docusaurus-plugin-content-docs/src/metadata.ts

```diff
@@ -101,6 +101,14 @@ export default async function processMetadata({
     throw new Error('Document id cannot include "/".');
   }
 
+  const isDocsHomePage = homePageId === baseID;
+
+  if (frontMatter.slug && isDocsHomePage) {
+    throw new Error(
+      `The docs homepage (homePageId=${homePageId}) is not allowed to have a frontmatter slug=${frontMatter.slug} => you have to chooser either homePageId or slug, not both`,
+    );
+  }
```

Comment:
@lex111 introduce a new warning, as discussed in https://github.com/facebook/docusaurus/issues/2879

As we don't duplicate the docs home (ie no /docs + /docs/introduction), then we should rather prevent the user to use the :slug frontmatter, as that wouldn't make sense)